broadcast/src/linked/mod.rs

```diff
@@ -0,0 +1,663 @@
+//! Secure, ordered broadcast protocol with linked messages for consistent network
+//! dissemination using cryptographic proofs, sequential linking, and robust state
+//! management.
+//!
+//! # Core Concepts
+//!
+//! - **Sequential Linking**: Every broadcast message contains a cryptographic reference
+//!   to the previous message, forming an immutable chain.
+//!
+//! - **Threshold Signatures**: Validators produce partial signatures that combine into a
+//!   threshold signature, ensuring a quorum verifies each message.
+//!
+//! - **Epochs**: The protocol is divided into epochs. Each epoch defines the active set of
+//!   signers and sequencers for orderly transitions and consensus.
+//!
+//! - **Storage**: Nodes persist state changes (links, acknowledgements) to a journal.
+//!   This enables reliable recovery and prevents conflicts like duplicate messages.
+//!
+//! # Design
+//!
+//! The system has two sets of nodes: Sequencers and Validators. The sets may overlap.
+//! Sequencers link messages, while Validators verify and sign them. Each message includes a
+//! threshold signature over its predecessor for an unbroken, verifiable chain.
```

Comment:
I think we should stay away from all these qualifiers on `chain`. Before it is "immutable chain" and here it is "unbroken, verifiable chain". I think it cheapens the writing a bit.